src/platform/qt/ScriptingView.cpp

```diff
@@ -57,27 +60,20 @@ void ScriptingView::load() {
 	}
 }
 
-void ScriptingView::addTextBuffer(ScriptingTextBuffer* buffer) {
-	QTextDocument* document = buffer->document();
-	m_textBuffers.append(buffer);
-	QListWidgetItem* item = new QListWidgetItem(document->metaInformation(QTextDocument::DocumentTitle));
-	connect(buffer, &ScriptingTextBuffer::bufferNameChanged, this, [item](const QString& name) {
-		item->setText(name);
-	});
-	connect(buffer, &QObject::destroyed, this, [this, buffer, item]() {
-		m_textBuffers.removeAll(buffer);
-		delete item;
-	});
-	m_ui.buffers->addItem(item);
-	m_ui.buffers->setCurrentItem(item);
+void ScriptingView::controllerReset() {
+	selectBuffer(QModelIndex());
+}
+
+void ScriptingView::bufferAdded(const QModelIndex&, int row, int) {
```

Comment:
I really wish there was a better way to do _this_. This wrapper feels silly. But as far as I can tell, this is the best it can get.

Reply:
Might I recommend a lambda?

Reply:
I was on the fence about a lambda when I wrote it (I prefer avoiding lambdas that aren't closing over local variables, since you could have used a compile-time function instead of a run-time object), but if you're recommending it, then I think that tips the scales.